src/cascadia/QueryExtension/ExtensionPalette.cpp

```diff
@@ -373,6 +373,23 @@ namespace winrt::Microsoft::Terminal::Query::Extension::implementation
         _queryBox().Focus(FocusState::Programmatic);
     }
 
+    void ExtensionPalette::_exportMessagesToFile(const Windows::Foundation::IInspectable& /*sender*/,
+                                                 const Windows::UI::Xaml::RoutedEventArgs& /*args*/)
+    {
+        std::wstring concatenatedMessages{};
+        for (const auto groupedMessage : _messages)
+        {
+            for (const auto chatMessage : groupedMessage)
+            {
+                concatenatedMessages += chatMessage.as<ChatMessage>()->MessageContent() + L"\n";
```

Comment:
I'd recommend splitting the concatenation up into an extra `concatenatedMessages += L"\n"`. This doesn't just avoid creating an intermediary of the `MessageContent` + newline, but also aids in debugging, because it's easier to step over each line with F10.